### DISCUSSION TITLE
Create a CLI command to generate middleware files.

### DISCUSSION THREAD

**Discussion by alfonsusac** [vercel/next.js#73511] [Ideas]
### Goals

1. Generate a middleware file using the CLI
2. Help reduce to consult the documentation by providing:
  a. where to place middleware.ts/js
  b. what the parameter types are and their return types 

### Non-Goals

1. Implementing parameters for custom middleware templates.
2. Generating middleware code without developer input (automatic editing).

### Background

It would help developer experience if we don't need to look up the docs for what the middleware function requries. It is merely an addition to the `next` CLI that is optional for developer to use. They can still write the middleware.ts file manually but It would be great for me to not have to look up the docs everytime I wanted to make the middleware file.

### Proposal

This feature can be implemented in two ways:
- Adding a new CLI command, such as `npm next add middleware`.
- Including it as part of the create-next-app template, similar to existing features like local fonts, favicons, and layouts.

**Comment by Arinji2** [comment]
I would love for this to be a feature.. i have the middleware documentation bookmarked at this point :D
A nice ``Using Middleware?`` option in create-next-app would be awesome.
